backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,202 @@
+package backtraceio.library.metrics;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Message;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import backtraceio.library.common.BacktraceMathHelper;
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.interfaces.Api;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.services.BacktraceHandlerThread;
+import backtraceio.library.services.BacktraceReportSender;
+
+public abstract class BacktraceEventsHandler extends Handler {
+
+    private final static transient String LOG_TAG = BacktraceEventsHandler.class.getSimpleName();
+
+    protected final BacktraceHandlerThread backtraceHandlerThread;
+
+    /**
+     * Number of events in the queue
+     */
+    public int getCount() {
+        return events.size();
+    }
+
+    /**
+     * Maximum number of events in store. If number of events in store hit the limit
+     * BacktraceMetrics instance will send data to Backtrace.
+     */
+    private int maxNumEvents = 350;
+
+    public void setMaxNumEvents(int maxNumEvents) {
+        this.maxNumEvents = maxNumEvents;
+    }
+
+    public int getMaxNumEvents() {
+        return this.maxNumEvents;
+    }
+
+    /**
+     * Time between retries if metrics submission fails
+     */
+    private final int timeBetweenRetriesMillis;
+
+    /**
+     * List of events in the event queue
+     */
+    ConcurrentLinkedDeque<Event> events = new ConcurrentLinkedDeque<Event>();
```

Comment:
shouldn't we make it private? It's by default private, but since we're using it everywhere, shouldn't we mark it `private` to keep code consistent?

Reply:
Will explicitly mark this private, thank you!